### PR TITLE
Combine reducers

### PR DESCRIPTION
**Pull request by bloderxd**
This PR is a proposal to combine reducers kotlin solution, please @vitorprado review it.

this is related to https://github.com/Reduks/Reduks/issues/9

### CODE REVIEW

**Comment by bloderxd** [comment]
Just remembering I also removed out folder from git tracking

File: src/main/kotlin/com/reduks/reduks/combine_reducers/CombineReducers.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.reduks.reduks.combine_reducers
+
+import com.reduks.reduks.Action
+
+/**
+ * Created by bloder on 18/02/18.
+ */
+
+class CombineReducers<State>(private val reducers: Array<out (State, Action<State>) -> State>) {
```
Comment (vitorprado) [inline src/main/kotlin/com/reduks/reduks/combine_reducers/CombineReducers.kt:9]: I think the best way to do this is using a fold:
```kotlin

class CombineReducers<State>(private val reducers: List<(State, Action<State>) -> State>) {

    constructor(vararg reducers: (State, Action<State>) -> State) : this(reducers.asList())

    fun enhance() : (State, Action<State>) -> State {
        return reducers.fold({ state, _ -> state }) { acc, reducer ->
            { state, action -> acc(reducer(state, action), action) }
        }
    }
}
```

File: src/main/kotlin/com/reduks/reduks/combine_reducers/CombineReducers.kt [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        var state: State = initialState
+        reducers.forEach {
+            val currentState = it(state, action)
+            state = if (currentState != state) currentState else state
+        }
+        return state
+    }
+}
+
+fun <State> Any.combineReducers(vararg reducers: (State, Action<State>) -> State) : (State, Action<State>) -> State = CombineReducers(reducers).combine()
```
Comment (vitorprado) [inline src/main/kotlin/com/reduks/reduks/combine_reducers/CombineReducers.kt]: This is useless IMHO

Comment (vitorprado) [reply src/main/kotlin/com/reduks/reduks/combine_reducers/CombineReducers.kt]: It can be a simple constructor.

**Review by vitorprado — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/reduks/reduks/combine_reducers/CombineReducers.kt (hunk 2 above)
Comment (bloderxd) [reply src/main/kotlin/com/reduks/reduks/combine_reducers/CombineReducers.kt]: I like to use extension here to look like redux 

Redux
```javascript
combineReducers({ reducer1,  reducer2 })
```

Reduks
```javascript
combineReducers(reducer1, reducer2)
```

We need to think about library design, could it be a simple constructor? yes, but how we discussed before, it's interesting we design the library like redux did

**Comment by bloderxd** [comment]
Please check it again @vitorprado

**Review by vitorprado — APPROVED** [review]
That's ok! 
But I prefer to use the constructor with `varargs` instead of `extensions` 
I think `extensions` should be used to extend a closed class not to add `combineReducers` to all classes.
I don't wanna users calling `combineReducers` in strange places and using with their classes.
IMHO

**Pull request merged**